packages/eslint-plugin/src/rules/no-styled-tagged-template-expression/__tests__/rule.test.ts

```diff
@@ -31,6 +32,8 @@ const replaceAlias = (str: string) =>
     .replace('styled.div', 'styled2.div')
     .replace('styled(Base)', 'styled2(Base)');
 
+const replaceType = (str: string) => str.replace('styled.div', 'styled.div<{color: string}>');
```

Comment:
Hey @liamqma can you help me to understand how we are testing that 
`styled.div<{color: string}>` doesn't become `styled.div`

Should we add a test case (input code/expected output) in https://github.com/atlassian-labs/compiled/blob/master/packages/eslint-plugin/src/rules/no-styled-tagged-template-expression/__tests__/rule.test.ts#L47

Reply:
We are testing `styled.div<{color: string}>` doesn't become `styled.div`, because
`flatMap((t) => [t, createTypedInvalidTestCase(t, replaceType, replaceType)])` creates a new test case for every test in the list.
e.g.
```
code: `
  import { styled } from '@compiled/react';

  styled.div\`color: blue\`;
`,
output: `
  import { styled } from '@compiled/react';

  styled.div({
    color: "blue"
  });
`,
```
becomes
```
code: `
  import { styled } from '@compiled/react';

  styled.div<{color: string}>\`color: blue\`;
`,
output: `
  import { styled } from '@compiled/react';

  styled.div<{color: string}>({
    color: "blue"
  });
`,
```